packages/sync-actions/test/business-units-sync.spec.js

```diff
@@ -0,0 +1,713 @@
+          },
+        ],
+      },
+    ])
+  })
+  test('should build `setStores` action but empty', () => {
+    const before = {
+      stores: [
+        {
+          typeId: 'store',
+          key: 'canada',
+        },
+        {
+          typeId: 'store',
+          key: 'usa',
+        },
+      ],
+    }
+
+    const now = { stores: [] }
+
+    const actual = businessUnitsSync.buildActions(now, before)
+    expect(actual).toEqual([
+      {
+        action: 'setStores',
+        stores: [],
+      },
+    ])
+  })
+
+  test('should build `addAssociate` action', () => {
+    const before = { associates: [] }
+    const now = {
+      associates: [
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'some-customer-id',
+          },
+        },
+      ],
+    }
+
+    const actual = businessUnitsSync.buildActions(now, before)
+    const expected = [
+      {
+        action: 'addAssociate',
+        associate: {
+          customer: {
+            typeId: 'customer',
+            id: 'some-customer-id',
+          },
+        },
+      },
+    ]
+    expect(actual).toEqual(expected)
+  })
+
+  test('should build `removeAssociate` action', () => {
+    const before = {
+      associates: [
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'some-customer-id',
+          },
+          associateRoleAssignments: [
+            {
+              associateRole: {
+                typeId: 'associate-role',
+                key: 'admin',
+              },
+              inheritance: 'Enabled',
+            },
+          ],
+        },
+      ],
+    }
+    const now = { associates: [] }
+
+    const actual = businessUnitsSync.buildActions(now, before)
+    const expected = [
+      {
+        action: 'removeAssociate',
+        customer: {
+          typeId: 'customer',
+          id: 'some-customer-id',
+        },
+      },
+    ]
+    expect(actual).toEqual(expected)
+  })
+
+  test('should build both `change-` and `removeAssociate` actions', () => {
+    const before = {
+      associates: [
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'remove',
+          },
+        },
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'keep',
+          },
+        },
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'remove2',
+          },
+        },
+      ],
+    }
+    const now = {
+      associates: [
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'new',
+          },
+        },
+        {
+          customer: {
+            typeId: 'customer',
+            id: 'keep',
+          },
+        },
+      ],
+    }
+
+    const actual = businessUnitsSync.buildActions(now, before)
+    const expected = [
+      {
+        action: 'changeAssociate',
+        associate: { customer: { id: 'new', typeId: 'customer' } },
+      },
```

Comment:
If there are no changes on this associate why is being triggered as a `changeAssociate`?